ratis-server/src/main/java/org/apache/ratis/statemachine/impl/SimpleStateMachineStorage.java

```diff
@@ -109,8 +113,9 @@ public void cleanupOldSnapshots(SnapshotRetentionPolicy snapshotRetentionPolicy)
 
     if (allSnapshotFiles.size() > snapshotRetentionPolicy.getNumSnapshotsRetained()) {
       allSnapshotFiles.sort(Comparator.comparing(SingleFileSnapshotInfo::getIndex).reversed());
-      List<File> snapshotFilesToBeCleaned = allSnapshotFiles.subList(
-              snapshotRetentionPolicy.getNumSnapshotsRetained(), allSnapshotFiles.size()).stream()
+      List<SingleFileSnapshotInfo> snapshotToBeCleaned = allSnapshotFiles.subList(
+          snapshotRetentionPolicy.getNumSnapshotsRetained(), allSnapshotFiles.size());
+      List<File> snapshotFilesToBeCleaned = snapshotToBeCleaned.stream()
```

Comment:
Since we will check if a snapshot file exists before deleting the md5, we don't need the index and the code can be simplified as below:
```java
      allSnapshotFiles.subList(numSnapshotsRetained, allSnapshotFiles.size())
          .stream()
          .map(SingleFileSnapshotInfo::getFile)
          .map(FileInfo::getPath)
          .forEach(snapshotPath -> {
        LOG.info("Deleting old snapshot at {}", snapshotPath.toAbsolutePath());
        final boolean deleted = FileUtils.deletePathQuietly(snapshotPath);
        if (deleted) {
          final File md5file = MD5FileUtil.getDigestFileForFile(snapshotPath.toFile());
          FileUtils.deleteFileQuietly(md5file);
        }
      });
```